<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

    <xacro:macro name="ros2_control_test_config">
        <ros2_control name="hardware_interface" type="system">
            <hardware>
                <plugin>dynamixel_ros_control/DynamixelHardwareInterface</plugin>
                <param name="debug">true</param>
                <param name="port_name">/dev/ttyUSB0</param>
                <param name="baud_rate">57600</param>
                <param name="torque_on_startup">true</param>
                <param name="torque_off_on_shutdown">false</param>
            </hardware>

            <joint name="joint_1">
                <param name="id">1</param>
                <param name="position_control_mode">extended_position</param>
                <command_interface name="position"/>
                <command_interface name="velocity"/>
                <state_interface name="position"/>
                <state_interface name="velocity"/>
                <state_interface name="effort"/>
            </joint>
        </ros2_control>
    </xacro:macro>
</robot>
